docs/swagger.yaml

```diff
@@ -0,0 +1,7643 @@
+swagger: '2.0'
```

Comment:
Are we sure we want to commit this to source control? Might be better to make it part of provisioning.

Reply:
I'm not wedded to committing this, especially since updating it could easily be forgotten. I don't know how other consumers of this file will get it.  When people want to generate client code (for example), where will they get this file from?

Reply:
That's true... we need a place for consumers to get the up-to-date `swagger.yaml`.

What about hitting `courses.edx.org/swagger.yaml`? Would that fit the use-cases you are thinking of?

If we do end up needing to commit this to source control, though, you could add a `validate_api_committed` build step that ensures that running `make api-docs` doesn't change `swagger.yaml`. Registrar does something like that currently.

Reply:
Visiting a URL like `courses.edx.org/swagger.yaml` won't work for all uses, because it will be subject to feature toggles.  Swagger doesn't attempt to provide a solution for "conditional endpoints".

If we're adding a build step, it should be to build and commit the file, not to check that it has been done right.

Reply:
Sure! As long as there's something other than hoping everyone remembers to manually generate it before committing.

Reply:
There's nothing yet.  People who don't run "make api-docs" won't dirty their working tree with a new swagger.yaml.  This is something still to be decided: how to build the docs.